src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ManagedIndexMetaData.kt

```diff
@@ -93,27 +123,13 @@ data class ManagedIndexMetaData(
         }
 
         val transitionToExists = transitionTo != null
-
         if (transitionToExists) {
             builder.field(TRANSITION_TO, transitionTo)
-        }
-
-        if (stateMetaData != null && !transitionToExists) {
-            builder.startObject(StateMetaData.STATE)
-            stateMetaData.toXContent(builder, params)
-            builder.endObject()
-        }
-
-        if (actionMetaData != null && !transitionToExists) {
-            builder.startObject(ActionMetaData.ACTION)
-            actionMetaData.toXContent(builder, params)
-            builder.endObject()
-        }
-
-        if (policyRetryInfo != null) {
-            builder.startObject(PolicyRetryInfoMetaData.RETRY_INFO)
-            policyRetryInfo.toXContent(builder, params)
-            builder.endObject()
+        } else {
+            builder.addObject(StateMetaData.STATE, stateMetaData, params)
+                .addObject(ActionMetaData.ACTION, actionMetaData, params)
+                .addObject(StepMetaData.STEP, stepMetaData, params)
+                .addObject(PolicyRetryInfoMetaData.RETRY_INFO, policyRetryInfo, params)
```

Comment:
This seems to alter the behavior since policyRetryInfo didnt have a check for transitionToExists

Reply:
yes, moved out